### PR TITLE
Feature/files 142 pdf preview

### PR DESCRIPTION
**Pull request by CMazzilli**

### CODE REVIEW

File: views/components/preview/PdfPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+		if (!show) {
+			setCurrentZoom(zoomStep[0]);
+			setIncrementable(true);
+			setDecrementable(false);
+			setFitToWidthActive(false);
+		}
+	}, [show]);
+
+	const increaseOfOneStep = useCallback(
+		(ev) => {
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:210]: implicit any

File: views/components/preview/PdfPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+					}
+				}
+			}
+			setFitToWidthActive(false);
+		},
+		[currentZoom, decrementable]
+	);
+
+	const fitToWidth = useCallback(
+		(ev) => {
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:250]: implicit any

File: views/components/preview/PdfPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	useEffect(() => {
+		if (show && fitToWidthActive) {
+			window.addEventListener('resize', fitToWidth);
+		}
+		return (): void => {
+			window.removeEventListener('resize', fitToWidth);
+		};
+	}, [fitToWidth, previewRef, show, fitToWidthActive]);
+
+	const resetWidth = useCallback((ev) => {
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:271]: implicit any

File: views/components/preview/PdfPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+					// onLoadSuccess={index === 0 ? onPageLoadSuccess : undefined}
+					width={currentZoom}
+					renderTextLayer={renderTextLayer}
+				/>
+			));
+		}
+		return [];
+	}, [currentZoom, numPages, renderTextLayer]);
+
+	const onDocumentLoadSuccess = useCallback((args) => {
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:299]: NonNullable<DocumentProps['onLoadSuccess']>

File: views/components/preview/PdfPreview.tsx [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+	Container,
+	IconButton,
+	Portal,
+	Tooltip,
+	useCombinedRefs
+} from '@zextras/carbonio-design-system';
+import findIndex from 'lodash/findIndex';
+import findLastIndex from 'lodash/findLastIndex';
+import map from 'lodash/map';
+import { Document, Page } from 'react-pdf/dist/esm/entry.webpack';
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:18]: import { DocumentProps, Document, Page } from 'react-pdf';

File: views/components/preview/PdfPreview.tsx [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+							{/*			size="medium" */}
+							{/*			backgroundColor="gray0" */}
+							{/*			iconColor="gray6" */}
+							{/*			borderRadius="round" */}
+							{/*		/> */}
+							{/*	</Padding> */}
+							{/* </Container> */}
+							<PreviewContainer ref={previewRef}>
+								{!$customContent ? (
+									<Document file={file} onLoadSuccess={onDocumentLoadSuccess}>
```
Comment (beawar) [inline views/components/preview/PdfPreview.tsx:374]: $customContent || (<Document ... >...</Document>)

File: views/components/preview/PreviewCriteriaAlternativeContent.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+export const PreviewCriteriaAlternativeContent: React.VFC<
+	PreviewCriteriaAlternativeContentProps
+> = ({ downloadSrc, openSrc }) => {
+	const ancRef = useRef<HTMLAnchorElement>(null);
+	const ancRef2 = useRef<HTMLAnchorElement>(null);
+
+	const downloadClick = useCallback(
+		(ev) => {
```
Comment (beawar) [inline views/components/preview/PreviewCriteriaAlternativeContent.tsx:36]: implicit any

File: views/components/preview/PreviewManager.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+				default: {
+					return state;
+				}
+			}
+		},
+		[]
+	);
+
+	const createPreview = useCallback(
+		({ onClose, ...props }) => {
```
Comment (beawar) [inline views/components/preview/PreviewManager.tsx:48]: implicit any

File: views/components/preview/PdfPreview.tsx (hunk 5 above)
Comment (CMazzilli) [reply views/components/preview/PdfPreview.tsx:18]: Followed https://github.com/wojtekmaj/react-pdf#configure-pdfjs-worker wiki

**Review by beawar — APPROVED** [review]

**Pull request merged**